### PR TITLE
Bump to webpack@2 and enable HMR

### PR DESCRIPTION
**Pull request by deepsweet**
* [x] `yarn start` (`webpack.config.js`)
* [x] `yarn test` (`webpack.config.test.js`)
* [x] split webpack config into `dev` and `build`

@xaviervia what do you think about splitting webpack config into 2: `dev` and `build`? right now that `index` entry point is slowing down initial dev-setup (and probably HMR) a lot.

### CODE REVIEW

**Comment by xaviervia** [comment]
@deepsweet for sure!

File: package.json [hunk 1]
```diff
@@ -9,8 +9,8 @@
   "license": "SEE LICENSE IN LICENSE",
   "main": "index-built.js",
   "scripts": {
-    "start": "webpack-dev-server --port 7777 --host 0.0.0.0 --history-api-fallback",
-    "build": "webpack",
+    "start": "cross-env NODE_ENV=development webpack-dev-server",
```
Comment (xaviervia) [inline package.json]: 👍

**Comment by xaviervia** [comment]
BTW don't worry about the Codeship integration, it's only for publishing the Showroom on `master`.

**Comment by deepsweet** [comment]
done, but not sure how to test `index-built.js` 🤔 looks legit.

**Comment by xaviervia** [comment]
Hm, interesting. Let me try it out.

**Comment by xaviervia** [comment]
Tried the built version by putting it in an HTML, it’s working. I’ll fix Codeship so that when we merge it publishes the Showroom.

**Comment by xaviervia** [comment]
Cool. Codeship fixed. Let's merge and release this soon.

**Pull request merged**